projects/scullyio/ng-lib/src/lib/transfer-state/transfer-state.service.ts

```diff
@@ -69,69 +58,40 @@ export class TransferStateService {
     /**
      * Each time the route changes, get the Scully state from the server-rendered page
      */
-    if (!isScullyGenerated()) return;
+    if (!isScullyGenerated()) {
+      return;
+    }
 
     this.router.events
       .pipe(
         filter(e => e instanceof NavigationStart),
-        tap(() => (this.fetching = new Subject<any>())),
```

Comment:
When you removed this, you created a race condition that is bad. 

This all fires on navigation start. If the route finishes loading before the http finishes, then the `getState` calls will get the previous route's data. This is bad. I think you need to add this timing affordance back into the code.

Reply:
I have rewritten it a bit to take this into consideration.
Also, it now will not emit a new value when the http is done before the navigation. The old situation would emit the new data to the old page, which is just as bad.

Reply:
Did another rewrite after our conversation.
getState will now never emit stale data. Neither will it get new data to an old page.
As soon as navigation begins, the emissions are suspended, until navigation **and** data fetching are completed.